env.go

```diff
@@ -78,6 +90,23 @@ func getProviderPath(path string) (*string, string) {
 		var fileProviderName = file.ProviderName
 		return &fileProviderName, strings.TrimPrefix(path, "file:")
 	}
+	if strings.HasPrefix(path, "vault:") {
+		var vaultProviderName = vault.ProviderName
+		// Do not remove the prefix since it will be processed during injection
+		return &vaultProviderName, path
+	}
 
 	return nil, path
```

Comment:
Let's just return an empty string here, so we can avoid a pointer. Maybe return a third bool argument to indicate that parsing the provider path was successful.